File: customview/src/main/java/com/cc/dc/customview/delegate/adapter/NewsAdapter.java
package com.cc.dc.customview.delegate.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import com.cc.dc.customview.delegate.bean.NewsEntity;
import com.hannesdorfmann.adapterdelegates3.AdapterDelegatesManager;

import java.util.List;

/**
 * Created by dc on 2018/2/28.
 * 新闻的Adapter
 */

public class NewsAdapter extends RecyclerView.Adapter {

    private Context context;
    private AdapterDelegatesManager<List<NewsEntity>> delegatesManager;

    private List<NewsEntity> list;


    public NewsAdapter(Context context, List<NewsEntity> list) {
        this.list = list;
        this.context = context;
        delegatesManager = new AdapterDelegatesManager<>();

        delegatesManager.addDelegate(new NewsNormalAdapter(context));
        delegatesManager.addDelegate(new NewsVideoAdapter(context));
    }

    @Override
    public int getItemViewType(int position) {
        return delegatesManager.getItemViewType(list, position);
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return delegatesManager.onCreateViewHolder(parent, viewType);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        delegatesManager.onBindViewHolder(list, position, holder);
    }

    @Override
    public int getItemCount() {
        return list.size();
    }
}

File: kotlindemo/src/main/java/com/cc/dc/kotlindemo/adapter/news/NewsAdapter.kt
package com.cc.dc.kotlindemo.adapter.news

import android.content.Context
import com.cc.dc.kotlindemo.adapter.base.BaseMultiTypeAdapter
import com.cc.dc.kotlindemo.bean.news.NewsEntity
import com.hannesdorfmann.adapterdelegates3.AdapterDelegatesManager

/**
 * Created by dc on 2018/5/30.
 * news adapter
 */
class NewsAdapter(context: Context, list: List<NewsEntity>) : BaseMultiTypeAdapter<NewsEntity>(context, list) {

    override fun initDelegateAdapter(delegatesManager: AdapterDelegatesManager<List<NewsEntity>>?) {
        delegatesManager!!.addDelegate(NewsNormalAdapter(context!!))
        delegatesManager!!.addDelegate(NewsImageAdapter(context!!))
        delegatesManager!!.addDelegate(NewsVideoAdapter(context!!))
    }
}